mag.h

```diff
@@ -112,6 +112,23 @@ _fmpz_sub2_fast(fmpz_t z, const fmpz_t x, const fmpz_t y, slong c)
     }
 }
 
+static __inline__ void
+_fmpz_2times_fast(fmpz_t res, const fmpz_t x, slong c)
+{
+    slong cx = *x;
+
+    if (!COEFF_IS_MPZ(*res) && (cx > COEFF_MIN / 2 && cx < COEFF_MAX / 2))
```

Comment:
These conditions are wrong. See _fmpz_add2_fast. Anyway, I think this function is not really needed; the compiler should be able to eliminate the redundant branches in _fmpz_add2_fast when passed the same operand twice.

Reply:
It has actually one property that I do not think the compiler can transform into, and that is that we now can allow `cx` to be up to size `COEFF_MAX / 2` instead of `COEFF_MAX / 4`.

Reply:
`_fmpz_add2_fast`/`_fmpz_sub2_fast` need to work for values of `c` larger than +/- 1. I think it's confusing if `_fmpz_2times_fast` in contrast to the other functions only supports `|c| <= 1`.

In practice, it makes no difference to go up to `COEFF_MAX / 2` instead of `COEFF_MAX / 4` since such huge exponents are very rare. We only care about optimizing for small exponents here. Better to be uniform and conservative.

Considering this, I don't really think this extra function is needed. 

Clearly these functions and their assumptions should be documented...

Reply:
Okay, will remove it then.